h/views/api_groups.py

```diff
@@ -27,8 +28,8 @@ def groups(request):
     all_groups = list_svc.request_groups(user=request.user,
                                          authority=authority,
                                          document_uri=document_uri)
-
-    all_groups = GroupsJSONPresenter(all_groups, links_svc).asdicts(expand=expand)
+    all_groups = [GroupResource(group, links_svc) for group in all_groups]
```

Comment:
This is, I'd wager, an interim step—I didn't want to go too much further on this branch without discussing things, but I had thought a route factory or somesuch here might make sense (i.e. the view would receive a list of the correct Group Resources as `context` instead of having to do the conversion itself).

Reply:
Yeah, if it can be done nicely I think the view receiving a list of group resources as context might be good. So that database lookups of model objects are handled by resources not by views. This may require creating a `GroupsResource` class which, as well as being a Pyramid context resource, is also an iterable of `GroupResource`s.